src/ui/components/JEXLDebuggerPage.tsx

```diff
@@ -10,17 +10,119 @@ import { Container, Row, Col, Form, Button } from "react-bootstrap";
 
 import { useToastsContext } from "../hooks/useToasts";
 import { evaluateJexl } from "../jexlParser";
+import { debounce } from "../utils/functional";
+
+type ContextValue = object | string | boolean | number | Date;
+type ContextValueString = "object" | "string" | "boolean" | "number" | "Date";
+
+type ContextFieldProps = {
+  contextKey: string;
+  onChange: (key: string, value: ContextValue) => void;
+  contextValue: ContextValue;
+  fieldType: ContextValueString;
+};
+
+const ContextField: React.FC<ContextFieldProps> = ({
+  contextKey,
+  contextValue,
+  onChange,
+  fieldType,
+}: ContextFieldProps) => {
+  const handleChange = useCallback(
+    (e: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => {
+      const newValue =
+        e.target.type === "checkbox"
+          ? (e.target as HTMLInputElement).checked
+          : e.target.value;
+      onChange(contextKey, newValue);
+    },
+    [contextKey, onChange],
+  );
```

Comment:
We can have a better typing story here.

onChange should take (key: string, value: string | boolean, fieldType: FieldType).

however, if fieldType === boolean, we always want value to be a boolean. Otherwise we want it to be a string.

We can accomplish this with overloads:

```ts
type OnChangeFn = {
    (key: string, value: boolean, fieldType: "boolean"): void;
    (key: string, value: string, fieldType: Exclude<FieldType, "boolean">): void;
};
```

We can also specialize the ContextField based on the type of value passed in:

```ts
type FieldType = "string" | "number" | "boolean" | "object" | "date";
type ContextFieldProps<TValue extends FormDataValue> = {
    // if TValue is boolean, this must be "boolean", otherwise its not "boolean"
    fieldType: TValue extends boolean ? "boolean": Exclude<FieldType, "boolean">;
    onChange: OnChangeFn;
    contextKey: string;
    value: FormDataValue;
};

function ContextField<TValue extends FormDataValue>({
  fieldType,
  onChange,
  contextKey,
  value,
}: ContextFieldProps<TValue>) {
// ...
}
```

You can see a full worked example at this playground [link](https://www.typescriptlang.org/play/?#code/JYWwDg9gTgLgBAKjgQwM5wEoFNkGN4BmUEIcARFDvmQNwBQdMAnmFnAGLQgAiyMyANWQAbAK5sAvHFQwowAHYBzOAB84AIwgRhOefWasOXXvzhTsuaABMAPDLlKANEag8+gkeIB8+lm3bAWMJWACp+ZuT2CopkquSa2rqxamTyoiDqWFDJ5BDqAFZY1HFkJli0jOEA8vIAwgAWyEpY7PIRAN50cN1wABQA1lhMAFzSstHOAG6eWKMJOk3OBIHBYayjZPNJAJSjkxDAVvQ9fYMjYw6KUzOjUU5wy0GhfqMAogAeuGJWWDYBT2ssM5NloFvIyF5dnB9od6ABfeiVQy1CDyGBYd4wf7BAAKxDAqBsISEYjYGPR8is6E4rhMHlJXg6XR6j1WLzgxJmcHJWEp6C2TTgAH54qCkm9Pt9ftjnqxgQLwT5md1UQ0mopZnAamrmq1jj1LGjyQBpIa3cZKfXdaak0ac0nwxEEUTyfDAVFwFFGzEyokk8TczG8qkuNz8f1YLy9TrdVmyoHM1WNZqOZmG9GY01MVPWmapuGjL0ZrErKx4iAEv0zSFwGMG1EyODJqw6HUaiLYPAwAB0olQWFqImE6jw-V6vWVJ25o07+G7bawr0mvJgNgAEiEALIAGQAkvIwKIYK8dCAV3ENzuQuSAIKUZAnrBntFeSfbMyMutTuDAAh9OOAmYEhSCCiRNGQ75ft+PRJuqWC9OmJpDM4vRYN2-BQBq8BoHAl57geR6Ps+MDbN2uD1EUgxWEspaAtsVpTnC3LCP2taTtBcCwc0CGosWWbOGhGFYd2NriDRAJ+PR7E9HCk5ws4ADaXEas4iGZshDy0X4AC6UknJOlAwKIUBtL0NheDYAD0kLwgwdAYpAsBwD8BDIKIwiEC6boegAUq8AAa27cFg6iiIoGpQDiyAatGBacQURQkWxJyGo2CkEMY7jOP2WKZfw2kdlQPZ9lgADK-Don8eXIFGUEnGQqBkBsFHCMIEBkDm0FkDATVwLIYlyXp9byI2ymSJgRW9v2g6tSOuD9DY2rJhqrRRtJ3Q8d6MBZualzXLaFzRHECriWy6wcFprDvhIn7rXA8lwApOU0mGyDaW+iInIZxmmXd5l3ScNhFuSMpwGp21DBIDWxABfhQ3cMScXUy1YBI7RjUxomo+0GW0u4CnQ9pTHWQDPRA7xIOlmDFPqUwUM9ZpEmsFDCqxGNaMY9CMxo7jr0Ez1RNwCTHEnO0QtIDAFEPNobUAO5HRiyDgDoP7oAoXyiD8Vh9RA0j1BAst9VLBhklAxBQN2pPdGuBtYMuUDZerbC83SqvOUw8hK8AuDOLLbCy7xcBYX14RZOb6AUZQcCUIo7nIHIwhMJbIs9AAQkecB+5nwCtUHWDwJLbCZPIWDLDA6AQH+JtgxR81HQoRtsOROfa5Y4CoueCCWbJKfdOTW2g+DWZQwAjjDl2o6BYJs8jcEc7PzSY9z7RkGgVgEGQxOvhxVnbz02x0HCQA)

Reply:
(This is optional)